segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DebugRecoveryProcessor.java

```diff
@@ -131,6 +141,66 @@ protected void recoverOperation(DataFrameRecord<Operation> dataFrameRecord, Oper
         }
     }
 
+    /**
+     * The overridden behaviour in this method is to choose between  using a DebugDataFrameReader and the regular DataFrameReader.
+     * The DebugDataFrameReader does not throw exception on finding duplicate entries in the DurableDataLog, which could be used by
+     * repair commands in Admin CLI to fix corrupted log with duplicate entries.
+     * @param metadataUpdater The OperationMetadataUpdater to use for updates.
+     * @return  the number of operations recovered.
+     * @throws Exception any other exception other than DataCorruptionException
+     */
+    @Override
+    protected int recoverAllOperations(OperationMetadataUpdater metadataUpdater) throws Exception {
```

Comment:
Question: why do we need this? If the `DebugDataFrameReader` has been tweaked to be able to tolerate reading operations with wrong sequence numbers, do we really need to modify this? Or the only purpose of overriding this is just to use `DebugDataFrameReader` instead of `DataFrameReader`? If so, maybe we can isolate the `DataFrameReader` instantiation and reuse the rest of the method implementation inheriting from `DataFrameReader`, right?

Reply:
this was done to mostly keep the changes localized in "DebugRecoveryProcessor", and not touching "RecoveryProcessor".. But can as well do as you said to abstract out the creation of FrameReader which would induce minimal changes.

Reply:
Correct, just re-use the rest of this method and only override the instantiation of either `DataFrameBuilder` or `DebugDataFrameBuilder`. Note that in the case of `DebugDataFrameBuilder` we should still be able to choose whether we want to tolerate inconsistent operation numbers or not, as you are doing now.